mhs/common/mhs_common/workflow/asynchronous_express.py

```diff
@@ -1,12 +1,72 @@
 """This module defines the asynchronous express workflow."""
 from typing import Tuple
 
+import requests
+import utilities.integration_adaptors_logger as log
+from comms import transmission_adaptor
+from utilities import timing
+
+from mhs_common.messages import ebxml_request_envelope, ebxml_envelope
+from mhs_common.state import persistence_adaptor, work_description
 from mhs_common.workflow import common_asynchronous
 
+logger = log.IntegrationAdaptorsLogger('ASYNC_EXPRESS_WORKFLOW')
+
 
 class AsynchronousExpressWorkflow(common_asynchronous.CommonAsynchronousWorkflow):
     """Handles the workflow for the asynchronous express messaging pattern."""
 
-    async def handle_outbound_message(self, message_id: str, interaction_details: dict, payload: str) \
-            -> Tuple[int, str]:
-        raise NotImplementedError()
+    def __init__(self, persistence_store: persistence_adaptor.PersistenceAdaptor,
+                 transmission: transmission_adaptor.TransmissionAdaptor, party_key: str):
+        self.persistence_store = persistence_store
+        self.transmission = transmission
+        self.party_key = party_key
+
+    async def handle_outbound_message(self, message_id: str, correlation_id: str, interaction_details: dict,
+                                      payload: str) -> Tuple[int, str]:
+        logger.info('0001', 'Entered async express workflow to handle outbound message')
+        wdo = work_description.create_new_work_description(self.persistence_store, message_id,
+                                                           work_description.MessageStatus.OUTBOUND_MESSAGE_RECEIVED)
+        await wdo.publish()
+
+        try:
+            # Call message packer to serialise message
+            interaction_details[ebxml_envelope.MESSAGE_ID] = message_id
+            interaction_details[ebxml_request_envelope.MESSAGE] = payload
+            interaction_details[ebxml_envelope.FROM_PARTY_ID] = self.party_key
+            interaction_details[ebxml_envelope.CONVERSATION_ID] = correlation_id
```

Comment:
Could this correlation id come from the logging context? Or would that be a bad practice to rely on?

Reply:
It felt like a bad practice to me, which is why I didn't do it.